src/main/java/org/opensearch/knn/index/codec/util/CodecBuilder.java

```diff
@@ -8,6 +8,7 @@
 import org.apache.lucene.codecs.Codec;
 import org.opensearch.index.mapper.MapperService;
 import org.opensearch.knn.index.codec.KNN910Codec.KNN910Codec;
```

Comment:
Why do we need to keep 910 codec around in this builder?

Reply:
After taking a second thought it may be a good point, customer can build  codec directly using no arg constructor as having such constructor is a requirement from core OS. In such case the only place that uses/requires codec factory is our own code, and we can simplify factory logic and return only the latest version of codec. I've pushed commit for this, @jmazanec15 please check if that is something you had in mind

Reply:
Makes sense to me.